spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java

```diff
@@ -830,4 +836,121 @@ private boolean determineSupportsRowNumberFunction(DataSource dataSource) {
 		}
 		return false;
 	}
+
+	/**
+	 * A {@link SqlPagingQueryProviderFactoryBean} specialization that overrides the {@code Oracle} paging
+	 * {@link SafeOraclePagingQueryProvider provider} with an implementation that properly handles sort aliases.
+	 * <p><b>NOTE:</b> nested within the aggregate DAO as this is the only place that needs this specialization.
+	 */
+	static class SafeSqlPagingQueryProviderFactoryBean extends SqlPagingQueryProviderFactoryBean {
```

Comment:
I decided to extend the base instead of copy/pasta the entire Batch 4.x impl. 
I then just hijack the returned provider and wrap it in a delegate that inspects its set fields.